src/components/flow/actions/callbrain/CallBrainForm.tsx

```diff
@@ -14,15 +15,36 @@ import { BrainInfo } from '../../../../store/flowContext';
 import { applyVueInReact } from 'veaury';
 // @ts-ignore
 import Unnnic from '@weni/unnnic-system';
+import TextInputElement, {
+  TextInputSizes,
+} from 'components/form/textinput/TextInputElement';
+import { StringEntry } from 'store/nodeEditor';
 const UnnnicIcon = applyVueInReact(Unnnic.unnnicIcon);
 
 export interface CallBrainFormProps extends ActionFormProps {
   brainInfo: BrainInfo;
+  entry: string;
 }
 
-export class BrainForm extends React.Component<CallBrainFormProps> {
+export interface CallBrainFormState {
+  entry: StringEntry;
+}
+
+export class BrainForm extends React.Component<
+  CallBrainFormProps,
+  CallBrainFormState
+> {
+  constructor(props: CallBrainFormProps) {
+    super(props);
+    this.state = { entry: { value: this.props.entry || '' } };
+    bindCallbacks(this, {
+      include: [/^handle/, /^on/],
+    });
+  }
   private handleSave(): void {
-    this.props.updateAction(propsToAction(this.props.nodeSettings, this.props));
+    const brainState = { ...this.props };
+    brainState.entry = this.state.entry.value;
+    this.props.updateAction(propsToAction(this.props.nodeSettings, brainState));
```

Comment:
Since the state is being saved in the action, the function `propsToAction` name should be modified